installscripts/dockerfiles/jenkins-ce/chefclient.sh

```diff
@@ -0,0 +1,46 @@
+#Spin wheel for visual effects
+spin_wheel()
+{
+    RED='\033[0;31m'
+    GREEN='\033[0;32m'
+    NC='\033[0m'
+
+    pid=$1 # Process Id of the previous running command
+    message=$2
+    spin='-\|/'
+    printf "\r$message...."
+    i=0
+
+    while ps -p $pid > /dev/null
+    do
+        #echo $pid $i
+        i=$(( (i+1) %4 ))
+        printf "\r${GREEN}$message....${spin:$i:1}"
+        sleep .05
+    done
+
+    wait "$pid"
+    exitcode=$?
+    if [ $exitcode -gt 0 ]
+    then
+        printf "\r${RED}$message....Failed${NC}\n"
+        exit
+    else
+        printf "\r${GREEN}$message....Completed${NC}\n"
+
+    fi
+}
+
+
+cd ~/jazz-installer/installscripts
+sudo docker cp cookbooks/. jenkins-server:/tmp/jazz-chef/cookbooks/
+sudo docker cp jenkinsplugins jenkins-server:/tmp/jazz-chef/
+# Running chef-client to execute cookbooks
+sudo docker exec -u root -i jenkins-server bash -c "chef install /tmp/jazz-chef/cookbooks/Policyfile.rb && chef export /tmp/jazz-chef/cookbooks/Policyfile.rb /tmp/jazz-chef/chef-export && cd /tmp/jazz-chef/chef-export && sudo chef-client -z"
+
+# Once the docker image is configured, we will commit the image.
+sudo docker commit -m "JazzOSS-Custom Jenkins container" jenkins-server jazzoss-jenkins-server
```

Comment:
launchscript will spin the docker and write all the necessary parameters to terraform vars. once the update jenkins configurations resource is done, then the configure jenkins docker resource will invoke this script. so at that time we do a docker copy for the updated cookbooks (with default.rb) and jazz-installer-vars.json to docker container and run the chef client.

Reply:
Right, so to summarize: we can't copy and run the cookbooks in the Dockerfile because we stand up the container before we configure the Chef cookbook variables later in the process.

Reply:
Yes.